File: java/2134. Minimum Swaps to Group All 1's Together II.java
class Solution {
    public int minSwaps(int[] nums) {
        int n = nums.length, size = 0, right = 0, count = 0, left = 0;
        for (int i = 0; i < n; i++) {
            size += nums[i];
        }
        while (right < size) count += 1-nums[right++];
        int answer = count;
        while (left < n) {
            count += 1-nums[(right++) % n];
            count -= 1-nums[left++];
            answer = Math.min(answer, count);
        }
        return answer;
    }
}

File: go/2134. Minimum Swaps to Group All 1's Together II.go
func minSwaps(nums []int) int {
	n, size, right, count, left := len(nums), 0, 0, 0, 0
	for _, v := range nums {
		size += v
	}
	for right < size {
		count += 1-nums[right]
		right++
	}
	answer := count
	for left < n {
		count += 1-nums[right%n]
		count -= 1-nums[left]
		right++
		left++
		answer = min(answer, count)
	}
	return answer
}